File: src/Neuralm.Application.Messages/Requests/AuthenticateRequest.cs
namespace Neuralm.Application.Messages.Requests
{
    public class AuthenticateRequest : Request
    {
        public string Username { get; }
        public string Password { get; }
        public string CredentialTypeCode { get; }

        public AuthenticateRequest(string username, string password, string credentialTypeCode)
        {
            Username = username;
            Password = password;
            CredentialTypeCode = credentialTypeCode;
        }
    }
}

File: src/Neuralm.Presentation.Web/src/messages/requests/AuthenticateRequest.ts
import Request from './Request';

/**
 * Represents the AuthenticateRequest class.
 */
export default class AuthenticateRequest extends Request {
  public username: string;
  public password: string;
  public credentialTypeCode: string;

  /**
   * Initializes the AuthenticateRequest class.
   * @param username The user name.
   * @param password The password.
   */
  constructor(username: string, password: string) {
    super();
    this.username = username;
    this.password = password;
    this.credentialTypeCode = 'Name';
  }
}
